libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -407,3 +410,35 @@ void ConstantEvaluator::endVisit(TupleExpression const& _tuple)
 	if (!_tuple.isInlineArray() && _tuple.components().size() == 1)
 		m_values[&_tuple] = evaluate(*_tuple.components().front());
 }
+
+void ConstantEvaluator::endVisit(FunctionCall const& _functionCall)
+{
+	using util::keccak256;
+	using util::h256;
+
+	auto const* builtinFunction = dynamic_cast<MagicVariableDeclaration const*>(ASTNode::referencedDeclaration(_functionCall.expression()));
+	if (!builtinFunction)
+		return;
+
+	auto const* functionType = builtinFunction->functionType(true);
+	solAssert(functionType);
+	switch (functionType->kind())
+	{
+		case FunctionType::Kind::ERC7201:
+		{
+			solAssert(_functionCall.arguments().size() == 1);
+			auto const* argument = dynamic_cast<Literal const*>(_functionCall.arguments()[0].get());
+			solAssert(argument);
+			ASTString argStringLiteral = argument->valueWithoutUnderscores();
+			auto value =
+				u256(keccak256(h256(u256(keccak256(argStringLiteral)) - 1))) &
```

Comment:
Shouldn't it be the `abi.encode` of the input?
```suggestion
                keccak256(abi.encode(uint256(keccak256(argStringLiteral)) - 1)) &
```

Reply:
Also, why the `u256` conversion here? Isn't better to work with bytes and only convert everything to `u256` when necessary?